mountpoint-s3/tests/fuse_tests/fork_test.rs

```diff
@@ -380,6 +384,130 @@ fn mount_scoped_credentials() -> Result<(), Box<dyn std::error::Error>> {
+        .arg(bucket)
+        .arg(mount_point)
+        .arg(format!("--region={region}"))
+        .arg(format!("--prefix={prefix}"))
+        .arg("--sse=aws:kms:dsse")
+        .arg(format!("--sse-kms-key-id={key_id}"))
+        .arg("--auto-unmount")
+        .arg("--foreground");
+    if let Some(credentials) = credentials {
+        cmd.env("AWS_ACCESS_KEY_ID", credentials.access_key_id())
+            .env("AWS_SECRET_ACCESS_KEY", credentials.secret_access_key())
+            .env("AWS_SESSION_TOKEN", credentials.session_token().unwrap());
+    }
+    let child = cmd.spawn().expect("unable to spawn child");
+    wait_for_mount("mountpoint-s3", mount_point.to_str().unwrap());
+    child
+}
+
+#[cfg(all(feature = "sse_kms", not(feature = "s3express_tests")))]
+fn erroneous_write(mount_point: &Path) {
+    let mut f = fs::File::create(mount_point.join("file.txt")).expect("can not open file for write");
+    let data = vec![0xaa; 32];
+    let write_result = f.write_all(&data);
+    assert!(
+        write_result.is_err(),
+        "should not be able to write to the file without proper sse"
+    );
+}
+
+#[cfg(all(feature = "sse_kms", not(feature = "s3express_tests")))]
+#[test]
+fn write_with_inexistent_key() {
+    let (bucket, prefix) = get_test_bucket_and_prefix("mount_and_check_log");
+    let key_id = "SOME_INVALID_KEY";
+    let mount_point = assert_fs::TempDir::new().expect("can not create a mount dir");
+    let child = mount_with_sse(&bucket, mount_point.path(), &prefix, key_id, None);
+    erroneous_write(mount_point.path());
+    let expected_log_line =
+        regex::Regex::new(r"^.*WARN.*KMS.NotFoundException.*Invalid keyId \\'SOME_INVALID_KEY\\'.*$").unwrap();
+    unmount_and_check_log(child, mount_point.path(), &expected_log_line);
+}
+
+#[cfg(all(feature = "sse_kms", not(feature = "s3express_tests")))]
+#[test]
+fn write_with_no_permissions_for_a_key() {
+    let sse_key = get_test_kms_key_id();
+    let log_line_pattern = format!("^.*WARN.*User: [^ ]* is not authorized to perform: kms:GenerateDataKey on resource: {sse_key} because no session policy allows the kms:GenerateDataKey action.*$");
+    let expected_log_line = regex::Regex::new(&log_line_pattern).unwrap();
+    let policy_with_no_kms_perms = r#"{"Statement": [
+        {"Effect": "Allow", "Action": ["s3:*"], "Resource": "*"}
+    ]}"#;
+    let credentials = tokio_block_on(get_scoped_down_credentials(policy_with_no_kms_perms));
+    let (bucket, prefix) = get_test_bucket_and_prefix("mount_and_check_log");
+
+    let mount_point = assert_fs::TempDir::new().expect("can not create a mount dir");
+    let child = mount_with_sse(&bucket, mount_point.path(), &prefix, &sse_key, Some(credentials));
+    erroneous_write(mount_point.path());
+    unmount_and_check_log(child, mount_point.path(), &expected_log_line);
+}
+
+#[cfg(all(feature = "sse_kms", not(feature = "s3express_tests")))]
+#[test]
+fn read_with_no_permissions_for_a_key() {
+    // create file
```

Comment:
when separating the test into sections, we could use scopes so its clear what set of code that the comment is relevant for.